python/ray/tune/trial.py

```diff
@@ -340,3 +391,48 @@ def __str__(self):
         if self.experiment_tag:
             identifier += "_" + self.experiment_tag
         return identifier.replace("/", "_")
+
+    def __getstate__(self):
+        """Memento generator for Trial.
+
+        Sets RUNNING trials to PENDING, and flushes the result logger.
+        Note this can only occur if the trial holds a DISK checkpoint.
+        """
+        if not self._checkpoint.storage == Checkpoint.DISK:
+            raise ValueError("Checkpoint cannot be in-memory.")
+        state = self.__dict__.copy()
+
+        pickle_data = {
+            "_checkpoint": self._checkpoint,
+            "config": self.config,
+            "custom_loggers": self.custom_loggers,
+            "resources": resources_to_json(self.resources),
+            "sync_function": self.sync_function
+        }
+        state["__data__"] = binary_to_hex(cloudpickle.dumps(pickle_data))
```

Comment:
It would be more readable to call cloudpickle.dumps individually instead of aggregating them into one data blob.